compositor_pipeline/src/pipeline/input/mp4.rs

```diff
@@ -116,10 +122,6 @@ impl Mp4 {
 
 impl Drop for Mp4 {
     fn drop(&mut self) {
-        if let Some(stop_video) = &self.stop_video_thread {
-            stop_video.store(true, std::sync::atomic::Ordering::Relaxed);
```

Comment:
If I understand correctly after this change Drop on Mp4 will happen before the drop on VideoReader, so we will start removing file before we even set should_close to true.